core/chaincode/lifecycle/deployedcc_infoprovider.go

```diff
@@ -103,6 +103,39 @@ func (vc *ValidatorCommitter) ChaincodeInfo(channelName, chaincodeName string, q
 	}, nil
 }
 
+// AllChaincodesInfo returns a list of deployed chaincodes info, including both new lifecycle and legacy chaincodes
+func (vc *ValidatorCommitter) AllChaincodesInfo(channelName string, sqe ledger.SimpleQueryExecutor) ([]*ledger.DeployedChaincodeInfo, error) {
+	sqes := &SimpleQueryExecutorShim{
+		Namespace:           LifecycleNamespace,
+		SimpleQueryExecutor: sqe,
+	}
+	ccQuery := &ExternalFunctions{Resources: vc.Resources}
+	namespaceDefs, err := ccQuery.QueryNamespaceDefinitions(sqes)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]*ledger.DeployedChaincodeInfo, 0)
+	for ccName := range namespaceDefs {
+		deployedccInfo, err := vc.ChaincodeInfo(channelName, ccName, sqe)
```

Comment:
`QueryNamespaceDefinitions()` returns listing only from `_lifecycle` namespace. So, `ChaincodeInfo()` would never execute line 94. 

Still, it may lead to duplicate entries for the chaincodes that are defined in both `_lifecycle` and `lscc` and hence, at line 132, the results form `legacyCCs` needs to be filtered out, if already appeared in `_lifecycle`

Reply:
QueryNamespaceDefinitions() returns only namespaces using new lifecycle (_lifecycle namespace). What use case may lead lscc and _lifecycle namespaces to have duplicate entries? When a legacy chaincode is upgraded to a new lifecycle chaincode?

Reply:
yes